### PR TITLE
Convert PurchaseFormDialog styles to vanilla-extract

### PR DESCRIPTION
**Pull request by alebedev**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4415?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4415** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4415?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4414** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4414?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by alebedev** [comment]
### Merge activity

* **Jun 27, 9:28 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4415) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: apps/store/src/components/ProductPage/PurchaseForm/PriceCalculatorDialog.tsx [hunk 1]
```diff
@@ -16,47 +17,20 @@ export const PriceCalculatorDialog = ({ children, header, isOpen, toggleDialog }
...
-  height: '100%',
-  overflow: 'auto',
-  padding: theme.space.md,
-  backgroundColor: 'transparent',
-  isolation: 'isolate',
-})
-
-const DialogClose = styled(Dialog.Close)({
-  zIndex: 1,
-})
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductPage/PurchaseForm/PriceCalculatorDialog.tsx:53]: nit
If we don't need `z-index` on the dialog button anymore maybe we can remove `isolation: isolate` from dialog content 🤔

**Review by guilhermespopolin — APPROVED** [review]
💅🏻

**Pull request merged**